### PR TITLE
#1485 - Transition to Replace Triggers

### PR DESCRIPTION
**Pull request by emimaesmith**
##### Checklist

- [x] `manage.py test` passes
- [x] tests are included
- [x] migrations are included
- [x] documentation is changed or added

### Affected app(s)

ingest
queue
recipe

### Description of change

We are in the process of replacing the existing trigger system in Scale v6. In order to provide a transitional release, these changes introduce configuring a target recipe for a Strike or Scan while still allowing the existing system to continue to function.

Changes:
Added an optional (for now - in Scale 6 this will become required) `recipe` object to the Strike and Scan configuration. This configuration matches the recipe name, as well as the data/media types of each  input so the recipe may be queued directly from file ingest. 

Also included is a fix for #1490

### CODE REVIEW

**Comment by emimaesmith** [comment]
Parse jobs are more complicated as they are kicked off when the scale file is saved. If we want to keep a transition period, we would need a way to tell if the file has been created by a standalone parse job or as part of a recipe. I am currently depending on the job_exe.recipe field to be populated if the parse job is coming from a recipe.

File: scale/recipe/definition/json/definition_v6.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
             'type': 'string',
         },
         'input': INTERFACE_SCHEMA,
+        # 'input_conditon': DATA_FILTER_SCHEMA,
```
Comment (JohnPTobe) [inline scale/recipe/definition/json/definition_v6.py]: Why are the filters commented out?

Comment (emimaesmith) [reply scale/recipe/definition/json/definition_v6.py]: That's a leftover that didn't get removed from my first idea of implementing this. Originally thought we would have needed conditions for each input, but this is not actually the case. I'll remove the commented out line.

File: scale/job/serializers.py [hunk 2]
```diff
@@ -6,11 +6,11 @@
 import rest_framework.serializers as serializers
 
 from job.models import Job
-from job.job_type_serializers import JobTypeBaseSerializerV5, JobTypeBaseSerializerV6
-from job.job_type_serializers import JobTypeSerializerV5
-from job.job_type_serializers import JobTypeRevisionBaseSerializer
-from job.job_type_serializers import JobTypeRevisionSerializerV5, JobTypeRevisionSerializerV6
-from job.job_type_serializers import JobTypeRevisionDetailsSerializerV6
+# from job.job_type_serializers import JobTypeBaseSerializerV5, JobTypeBaseSerializerV6
```
Comment (JohnPTobe) [inline scale/job/serializers.py]: Can these imports be moved back to the top? If not, we need to delete these commented out lines.

**Pull request merged**